gateway/sds_gateway/api_methods/tasks.py

```diff
@@ -24,6 +25,121 @@
+        directory: Directory to check space for (defaults to MEDIA_ROOT)
+
+    Returns:
+        bool: True if enough space is available, False otherwise
+    """
+    if directory is None:
+        directory = Path(settings.MEDIA_ROOT)
+
+    try:
+        total, used, free = shutil.disk_usage(directory)
+        available_space = free - DISK_SPACE_BUFFER
+    except (OSError, ValueError) as e:
+        logger.error(f"Error checking disk space for {directory}: {e}")
+        return False
+    else:
+        return available_space >= required_bytes
+
+
+def estimate_zip_size(files: list[File]) -> int:
+    """
+    Estimate the size of a zip file containing the given files.
+
+    Args:
+        files: List of File model instances
+
+    Returns:
+        int: Estimated size in bytes (includes compression overhead)
+    """
+    total_file_size = sum(file_obj.size for file_obj in files)
+
+    # Estimate zip overhead (headers, compression metadata, etc.)
+    # Rough estimate: 10% overhead for small files, 5% for large files
+    overhead_factor = 1.1 if total_file_size < 100 * 1024 * 1024 else 1.05
+
+    return int(total_file_size * overhead_factor)
+
+
+def cleanup_orphaned_zips() -> int:
+    """
+    Clean up orphaned/partial zip files that don't have corresponding database records.
+
+    Returns:
+        int: Number of files cleaned up
+    """
+    media_root = Path(settings.MEDIA_ROOT)
+    temp_zips_dir = media_root / "temp_zips"
+
+    if not temp_zips_dir.exists():
+        return 0
+
+    cleaned_count = 0
+
+    try:
+        # Get all zip files in the temp_zips directory
+        for zip_file in temp_zips_dir.glob("*.zip"):
+            # Check if there's a corresponding database record
+            try:
+                # Extract UUID from filename (format: uuid_filename.zip)
+                filename = zip_file.name
+                if "_" in filename:
+                    uuid_part = filename.split("_")[0]
+                    # Check if this UUID exists in the database
+                    exists = TemporaryZipFile.objects.filter(uuid=uuid_part).exists()
+                    if not exists:
+                        # No database record, this is an orphaned file
+                        zip_file.unlink()
+                        cleaned_count += 1
+                        logger.info(f"Cleaned up orphaned zip file: {zip_file}")
+            except (OSError, ValueError) as e:
+                logger.error(f"Error processing zip file {zip_file}: {e}")
+                # Try to delete the problematic file anyway
+                try:
+                    zip_file.unlink()
+                    cleaned_count += 1
+                    logger.info(f"Cleaned up problematic zip file: {zip_file}")
+                except OSError:
+                    logger.error(f"Failed to delete problematic zip file: {zip_file}")
+
+    except OSError as e:
+        logger.error(f"Error during orphaned zip cleanup: {e}")
```

Comment:
This outer try-except block is not needed. `OSError` is already handled in the inner block and `.glob()` suppresses those exceptions, so this except will never run.